src/logicLayer/GameLogic.java

```diff
@@ -70,11 +74,70 @@ private boolean checkIfUnitIsOnTile(Tile tile) {
 		return tile.hasUnit();
 	}
 	
-	private int calculateMovCostBetweenTile(Tile startTile, Tile finalTile) {
-		int xDiff = Math.abs(startTile.getPos().getXPosition() - finalTile.getPos().getXPosition());
-		int yDiff = Math.abs(startTile.getPos().getYPosition() - finalTile.getPos().getYPosition());
-		int moveCost = xDiff+yDiff;
-		return moveCost;
+	private List<Tile> calculateValidTileToMoveTo(Unit unit) {
+		List<Tile> tileRange = new ArrayList<Tile>();
+		Direction direction = Direction.NONE;
+		Tile startTile = unit.getTile();	
+		int movStamina = unit.getStats().getMov().getCurrentValue()+1;
+		tileRange.add(startTile);
+		calculateMovRange(tileRange, startTile, movStamina, direction);
+		System.out.println();
+		return tileRange;
+	}
+	
+	private void calculateMovRange(List<Tile> tileRange, Tile startTile, int movStamina, Direction direction) {
+		if (movStamina == 0) {
+			return;

```

Comment:
This is in response too Rowan's earlier comment that was removed. Basically this if statement is to indicate our base case since this function is recursive. This is to help with readibility.